### PR TITLE
Sysbox doc improvements.

### PR DESCRIPTION
**Pull request by ctalledo**
A few more documentation improvements (see each of the commits).

### CODE REVIEW

File: docs/user-guide/features.md [hunk 1]
```diff
@@ -47,10 +34,47 @@
...
+
+-   Run Systemd inside a Docker container or K8s pod easily, without complex
+    container configurations.
+
+-   Enables you to containerize apps that rely on Systemd (e.g., legacy apps).
+
+## Docker-in-Docker
+
+-   Run Docker inside a container or pod easily without insecure privileged
+    containers or host Docker socket mounts.
```
Comment (rodnymolina) [inline docs/user-guide/features.md:47]: "host's"

File: docs/user-guide/features.md [hunk 2]
```diff
@@ -47,10 +34,47 @@
...
+
+-   Full isolation between the Docker inside the container and the Docker on the
+    host.
+
+-   The inner Docker container images can live ephemerally inside the container
+    or be cached on the host.
+
+## Kubernetes-in-Docker [ v0.2.0+ ]
+
+-   Use containers are Kubernetes nodes, instead of more expensive VMs.
```
Comment (rodnymolina) [inline docs/user-guide/features.md:57]: s / are / as

**Review by rodnymolina — APPROVED** [review]
LG

File: docs/user-guide/features.md (hunk 1 above)
Comment (ctalledo) [reply docs/user-guide/features.md]: Done.

File: docs/user-guide/features.md (hunk 2 above)
Comment (ctalledo) [reply docs/user-guide/features.md]: Done.

**Pull request merged**